### DISCUSSION TITLE
binding on a UserControl does not work

### DISCUSSION THREAD

**Discussion by mknotzer** [unoplatform/uno#6323] [Q&A]
[MyApp.zip](https://github.com/unoplatform/uno/files/6693577/MyApp.zip)
Uno 3.7.4

```
<controls:MyControl Text="{Binding Caption}"> <-- bug -->

<TextBlock Text="{Binding Caption}"/> <-- ok -->

using System;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;

namespace MyApp.Controls
{
	public sealed partial class MyControl : UserControl
	{
		// Text

		public static readonly DependencyProperty TextProperty =
			DependencyProperty.Register("Text", typeof(string), typeof(MyControl), new PropertyMetadata(default(string)));

		public string Text
		{
			get { return (string)GetValue(TextProperty); }
			set { SetValue(TextProperty, value); }
		}

		public MyControl()
		{
			InitializeComponent();

			DataContext = this;
		}
	}
}

```

**Comment by jeromelaban** [comment]
Thanks for the report!

This is an expected behavior. The reason for this is that the `DataContext` on `MyControl` is overriden by `DataContext = this`, which makes all bindings on that control evaluate from `typeof(MyControl)` and not `typeof(MainPage)`, therefore not finding a `Caption` property.

It's best to avoid modifying the `DataContext` property from the code behind (except from the root, or from the XAML itself), particularly from controls as it makes bindings change like in this example.

You can validate that the behavior is the same by running this sample from UWP.

If you have a design question for an application, we invite you to ask the question on the discussions section of this repo.
